core/src/main/java/org/apache/calcite/rel/rules/FilterJoinRule.java

```diff
@@ -84,6 +84,11 @@ protected void perform(RelOptRuleCall call, @Nullable Filter filter,
       return;
     }
 
+    if (filter != null
```

Comment:
This change is somewhat radical, since it will prevent pushing any predicates of the filter.
I wonder if the rule could be changed so that predicates with correlated variables cannot be pushed, while the others still can (assuming that the filter predicate is a conjunction of simpler predicates).

Reply:
I agree, I'll submit another PR later to make it even better.

Reply:
Here is how we customized another Calcite rule to do something similar: https://github.com/feldera/feldera/blob/main/sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/calciteCompiler/optimizer/FilterCorrelateRule.java

Reply:
It's here #4675. Mihai has already submitted the optimization to Calcite.